vamp-json/VampJson.h

```diff
@@ -1109,11 +1109,14 @@ class VampJson
 
     static bool
     successful(json11::Json j, std::string &err) {
-        if (!j["result"].is_object()) {
-            err = "result object expected for success";
+        if (j["result"].is_object()) {
+            return true;
+        } else if (j["error"].is_object()) {
+            return false;
+        } else {
```

Comment:
That would work. Or something like

```c++
    if (j["result"].is_object() && j["error"].is_object()) {
        err = "valid response may contain only one of result and error objects";
        return false;
    } else if (j["error"].is_object()) {
        return false;
    } else if (j["result"].is_object()) {
        return true;
    } else {
        err = "either a result or an error object is required for valid response";
        return false;
    }
```
which has the marginal advantage of providing distinct error messages for the two protocol-error cases. I suppose that could be made tidier with hasError and hasResult temporaries or something.